http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala

```diff
@@ -187,138 +124,330 @@ class GKEInterpreter[F[_]: Parallel: ContextShift](
+                                kubeSubNetwork)
+
+  override def pollCluster(params: PollClusterParams)(implicit ev: ApplicativeAsk[F, AppContext]): F[Unit] = {
     for {
-      clusterOpt <- kubernetesClusterQuery.getMinimalClusterById(clusterResult.clusterId).transaction
+      ctx <- ev.ask
+      _ <- logger.info(
+        s"Polling cluster creation for cluster ${params.clusterId} | trace id: ${ctx.traceId}"
+      )
+
+      // Grab records from the database
+      clusterOpt <- kubernetesClusterQuery.getMinimalClusterById(params.clusterId).transaction
       dbCluster <- F.fromOption(
         clusterOpt,
-        KubernetesClusterNotFoundException(s"cluster with id ${clusterResult.clusterId} not found in database")
+        KubernetesClusterNotFoundException(
+          s"Failed kubernetes cluster creation. Cluster with id ${params.clusterId} not found in database"
+        )
       )
       defaultNodepool <- F.fromOption(dbCluster.nodepools.filter(_.isDefault).headOption,
                                       DefaultNodepoolNotFoundException(dbCluster.id))
+
+      // Poll GKE until completion
       lastOp <- gkeService
-      //TODO: refactor this to handle interupts
         .pollOperation(
-          KubernetesOperationId(dbCluster.googleProject, dbCluster.location, clusterResult.operation),
+          params.createResult.op,
           config.monitorConfig.clusterCreate.interval,
           config.monitorConfig.clusterCreate.maxAttempts
         )
         .compile
         .lastOrError
-      _ <- if (lastOp.isDone) logger.info(s"Create cluster operation has finished for cluster ${dbCluster.id}")
+
+      _ <- if (lastOp.isDone)
+        logger.info(
+          s"Create cluster operation has finished for cluster ${params.clusterId} | trace id: ${ctx.traceId}"
+        )
       else
-        logger.error(s"Create cluster operation has failed for cluster ${dbCluster.id}") >> F.raiseError[Unit](
-          ClusterCreationException(
-            s"Failed to poll cluster creation operation to completion for cluster ${dbCluster.id} and default nodepool ${defaultNodepool.id}"
+        logger.error(
+          s"Create cluster operation has failed for cluster ${params.clusterId} | trace id: ${ctx.traceId}"
+        ) >>
+          // Note LeoPubsubMessageSubscriber will transition things to Error status if an exception is thrown
+          F.raiseError[Unit](
+            ClusterCreationException(
+              s"Failed to poll cluster creation operation to completion for cluster ${params.clusterId}"
+            )
+          )
+
+      // Resolve the cluster in Google
+      googleClusterOpt <- gkeService.getCluster(dbCluster.getGkeClusterId)
+      googleCluster <- F.fromOption(
+        googleClusterOpt,
+        ClusterCreationException(
+          s"Cluster not found in Google: ${dbCluster.getGkeClusterId} | trace id: ${ctx.traceId}"
+        )
+      )
+
+      // Install nginx igress controller
```

Comment:
Would it be better to move nginx ingress controller installation and monitoring to own method(s)?